### PR TITLE
add log file

### PR DESCRIPTION
**Pull request by chylli**

### CODE REVIEW

File: lib/App/Base.pm [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package App::Base;
 use strict;
 use warnings;
-our $VERSION = "0.06";
+our $VERSION = "0.05";
```
Comment (huiwang85) [inline lib/App/Base.pm]: Why downgrade version?

File: lib/App/Base/Daemon.pm [hunk 2]
```diff
@@ -271,14 +299,15 @@ sub __run {
...
+                # close all STD* files, and redirect STD* to log-file
                 for ( 0 .. 2 ) {
                     POSIX::close($_) unless $pid and $_ == $pid->fileno;
                 }
                 (         open( STDIN, '<', '/dev/null' )
-                      and open( STDOUT, '>', '/dev/null' )
-                      and open( STDERR, '>', '/dev/null' ) )
+                      and open( STDOUT, '>>', $self->log_file )
+                      and open( STDERR, '>>', $self->log_file ) )
                   or die "Couldn't open /dev/null: $!";
```
Comment (huiwang85) [inline lib/App/Base/Daemon.pm]: This `die` message should be updated accordingly.

File: lib/App/Base/Daemon.pm [hunk 3]
```diff
@@ -171,6 +175,24 @@ sub _build_pid_file {
...
+has log_file => (
+    is      => 'ro',
+    lazy    => 1,
+    builder => '_build_log_file',
+);
+
+sub _build_log_file {
+    my $self = shift;
+    my $file = $self->getOption('log-file') // '/dev/null';
+    return $file;
```
Comment (huiwang85) [inline lib/App/Base/Daemon.pm]: Should we check here to see if the specified file is writable?

File: lib/App/Base.pm (hunk 1 above)
Comment (chylli) [inline lib/App/Base.pm]: ghost did it ? I don't know either :)

File: lib/App/Base/Daemon.pm (hunk 2 above)
Comment (chylli) [inline lib/App/Base/Daemon.pm]: Weird. I thought I updated it already.

**Pull request closed without merging**